ambry-replication/src/main/java/com.github.ambry.replication/ReplicaThread.java

```diff
@@ -229,20 +219,51 @@ public void run() {
   }
 
   /**
-   * Replicas from the given replicas
-   * @param replicasToReplicate list of {@link RemoteReplicaInfo} by data node
+   * Do a lazy removal of {@link RemoteReplicaInfo} from current {@link ReplicaThread}.
+   * @param remoteReplicaInfo {@link RemoteReplicaInfo} to remove.
+   */
+  void removeRemoteReplicaInfo(RemoteReplicaInfo remoteReplicaInfo) {
+    lock.lock();
+    replicaLazyRemoveSet.add(remoteReplicaInfo);
+    lock.unlock();
```

Comment:
Every call to unlock needs to be in a finally block.
Even better, if it's a simple mutex lock, can switch to using synchronized(someLockObject) which handles unlocking automatically.

Reply:
There is a critical section in `replicate`. It's not suitable to use `synchronized ` for its case.

`finally` is used.